### ISSUE TITLE
logging output during tests

### ISSUE THREAD

**Issue by randy3k** [randy3k/UnitTesting#41]
@niosus wrote in https://github.com/randy3k/UnitTesting-example/issues/3

> I would like to see debug output during the run of my tests. Currently I can just see if the tests failed or not and the appropriate values in case they failed.

> Is it possible to show debug output always? Either output from print statements or, preferably using logging?

> Thanks!

**Comment by niosus** [comment]
oh, sorry, opened in a wrong repo. I should stop coding late. Thanks for moving it here!

**Comment by randy3k** [comment]
It should be possible by exposing the [stream](https://github.com/randy3k/UnitTesting/blob/d4b4fec5c9ac2e507a8b0adeee445bdaf55bd228/unittesting/common/test.py#L147-L147) object to users.

**Comment by randy3k** [comment]
6708fc0 added a logger, you could use logger in your `test*.py` as follows

```python
logger = logging.getLogger('UnitTesting')
logger.setLevel(logging.DEBUG)

class TestUnitTesting(TestCase):
     def test_something(self):
          logger.debug("hello")
```

**Comment by randy3k** [comment]
looks like ST2 does not like it...

**Comment by niosus** [comment]
Thanks @randy3k that was fast! Are you planning to make it part of the next version in the Package Manager? Or should I just clone this repo?

**Comment by randy3k** [comment]
It is released. You should be able to upgrade it via Package Control.

**Comment by niosus** [comment]
I guess I will have to wait some time till it becomes available through Package Control. I believe there is some lag till it is ready for the public. I have used PackageResourceViewer to verify there is still no logger present on my local machine. By the way, is there an easy way to check version of your plugin?

**Comment by randy3k** [comment]
PC has [updated](https://packagecontrol.io/packages/UnitTesting) UnitTesting, the latest version is 0.5.9.

You can use `Package Control: List Packages` to check package version.

**Comment by niosus** [comment]
Ah, you're totally right! Thanks for your patience :)

Logging something in tests works now. Do you think it is possible to also output something that is being logged from the package under test?

**Comment by randy3k** [comment]
You can use the same logger in your package.

**Comment by niosus** [comment]
Yes, I see this option, but my plugin has already nicely configured loggers with their specific names for each module, so in the ideal case I would still like to see them with their specific names. However, if you estimate that this is hard to do - it is not too crucial to me. Just a "nice to have" feature. Thanks anyway!

**Comment by randy3k** [comment]
You can configure your loggers such that all messages would be forward to the UnitTesting logger.

**Comment by randy3k** [comment]
@niosus 

The `logger.getLogger("UnitTesting")` is now deprecated. A new settings `capture_console` in [unittesting.json](https://github.com/randy3k/UnitTesting/blob/master/unittesting.json#L8) will redirect the sublime console to the output stream. It is still new, use with care. :)

**Comment by niosus** [comment]
Thanks! This sounds pretty awesome!

**Comment by niosus** [comment]
I think the setting is wrong. It works if I use `caputre_console` instead of `capture_console`.

Also, it doesn't seem to capture the outputs of `logging`. At least in my case. It prints the lines printed with `print(...)`

**Comment by randy3k** [comment]
The setting name has been fixed. It seems that logging is still using the original stdout even the the current stdout has been directed to the output stream.

**Comment by randy3k** [comment]
I have an idea how to fix it. I am out now, will fix when I get back home.

**Comment by randy3k** [comment]
fix in e3a3235d1ffc3867eec7d67ab1547c8e19312f63.

**Comment by niosus** [comment]
Perfect! Now it works exactly as I was dreaming :+1: Great job!
